backbone/src/main/java/org/researchstack/backbone/task/Task.java

```diff
@@ -40,6 +46,53 @@ public Task(String identifier)
     }
 
     /**
+     * Sets the current @{@link ViewTaskActivity} for this task.
+     * This function should be called only by the {@link ViewTaskActivity} instance running
+     * at its creation time.
+     * @param activity the @{@link ViewTaskActivity} instance
+     */
+    public void setActivity(ViewTaskActivity activity){
+        this.activity = activity;
+        activity.setStepViewCallback(new StepViewCallback() {
+            @Override
+            public void onStepShown(ViewTaskActivity activity, Step step) {
+                onStepChanged(activity, step);
+            }
+        });
+    }
+
+    /**
+     * Retrieves the @{@link ViewTaskActivity} that is actually running this task.
+     * Tasks can use this to have access to the current activity and its context.
+     */
+    public ViewTaskActivity getActivity(){
+        return this.activity;
+    }
+
+    /**
+     * Forces the task to proceed to a given @{@link Step}.
+     * Results, if any, should be computed before.
+     * @param step the @{@link Step}
+     */
+    public void forceStep(@NonNull Step step){
```

Comment:
getStepAfter/Before is called by the activity (when, e.g. a button is pressed).
forceStep() is "mandated" by the task.